File: dotnet/windntrees.core/DataAccess.Core/ModelsMetaData/Product/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;
using SharedLibrary.Core.Attributes;

namespace DataAccess.Core.Models
{
    [ModelMetadataType(typeof(ProductMetaData))]
    public partial class Product
    {

    }

    public partial class ProductMetaData
    {
        [Key]
        public System.Guid Uid { get; set; }

        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Name")]
        public string Name { get; set; }

        [LocaleMessageStringLength(1024)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Description")]
        public string Description { get; set; }

        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Version")]
        public string Version { get; set; }

        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Code")]
        public string Code { get; set; }

        [LocaleMessageStringLength(100)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "LegalCode")]
        public string LegalCode { get; set; }

        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "ProductYear")]
        public Nullable<int> ProductYear { get; set; }

        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Category")]
        public string Category { get; set; }

        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Manufacturer")]
        public string Manufacturer { get; set; }

        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Unit")]
        public string Unit { get; set; }

        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Color")]
        public string Color { get; set; }

        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Service")]
        public Nullable<bool> Service { get; set; }

        [LocaleMessageStringLength(50)]
        [Display(ResourceType = typeof(LocaleResources.Core.Contents.Product.Product), Name = "Picture")]
        public string Picture { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }

        public virtual LicenseInfo LicenseInfo { get; set; }
        public virtual ICollection<ProductFeature> ProductFeatures { get; set; }
    }
}

File: dotnet/windntrees.core/Application.Core/wwwroot/js/models/Product.js
function Product(data)
{
    var instance = this;

    instance._datakey = data.Uid;

    instance.Uid = data.Uid;

    instance.Name = ko.observable(data.Name);
    instance.Name.extend({
        required: true,
        maxLength: 100
    });

    instance.Description = ko.observable(data.Description);
    instance.Description.extend({
        maxLength: 1024
    });

    instance.Version = ko.observable(data.Version);
    instance.Version.extend({
        maxLength: 100
    });

    instance.Code = ko.observable(data.Code);
    instance.Code.extend({        
        maxLength: 100
    });

    instance.LegalCode = ko.observable(data.LegalCode);
    instance.LegalCode.extend({
        maxLength: 100
    });

    instance.ProductYear = ko.observable(data.ProductYear);    

    instance.Manufacturer = ko.observable(data.Manufacturer);
    instance.Category = ko.observable(data.Category);
    instance.Unit = ko.observable(data.Unit);
    instance.Color = ko.observable(data.Color);

    instance.Service = ko.observable(data.Service);

    instance.Picture = ko.observable(data.Picture);
    instance.RowVersion = data.RowVersion;


    instance.BriefName = ko.computed(function () {
        if (instance.Name() !== null && instance.Name() !== undefined) {

            if (instance.Name().length > 50) {

                return (instance.Name().substring(0, 17) + " ...");
            }

            return instance.Name();
        }
    });

    instance.PictureLink = ko.computed(function () {
        return "../../uploads/product/" + instance.Picture();
    });
}